window_manager/classes/storage.js

```diff
@@ -38,6 +45,62 @@ export class Storage {
     }
     Storage.#instance = this;
     console.log('Storage instance created');
+
+    /** @type {Promise<ValidatedConfiguration>} */
+    this.validatedConfiguration = Storage.loadConfigFromStorage();
+  }
+
+  /**
+   * Load config from storage. First try session storage, then read from synced.
+   * @return {Promise<ValidatedConfiguration>}
+   */
+  static async loadConfigFromStorage() {
+    const validConfig = /** @type {ValidatedConfiguration} */(
+      (await chrome.storage.session.get({[VALID_CONFIG_KEY]: null}))[VALID_CONFIG_KEY]);
+
+    if (validConfig?.valid) {
+      console.info(`${new Date().toLocaleTimeString()} Valid config read from session storage`);
+
+      // convert raw JS objects to classes.
+      validConfig.actions = StorageFromJson.actionsFromObj(validConfig.actions);
+      validConfig.matchers = StorageFromJson.matchersFromObj(validConfig.matchers);
+      validConfig.settings = StorageFromJson.settingsFromObj(validConfig.settings);
+      return validConfig;
+    }
+    // config in session store not present or not valid, try synced storage.
+    return Storage.loadConfigFromSyncedStorage();
+  }
+
+  /**
+   * Read config from synced storage, and update session storage copy.
+   * @return {Promise<ValidatedConfiguration>}
+   */
+  static async loadConfigFromSyncedStorage() {
+    const parsedConfig = Storage.parse((await Storage.getRawConfiguration()));
+
+    if (parsedConfig.valid) {
+      // update session storage.
+      chrome.storage.session.set({[VALID_CONFIG_KEY]: parsedConfig});
+      console.info(`${new Date().toLocaleTimeString()} Config read from synced storage, copied to session storage`);
+      return parsedConfig;
+    }
+
+    // Should never get here because synced storage is perfect in every way!
+    await chrome.storage.session.remove(VALID_CONFIG_KEY);
+    return Promise.reject(
+        new Error(`${new Date().toLocaleTimeString()
```

Comment:
isn't there something wrong with spaces around {}?

Reply:
It does look odd, but that ESLint's formatting style...  (which could be a bug in eslint!)

The other option is to put it all on one line...

Reply:
> isn't there something wrong with spaces around {}?

yeah. This is ESlint's idea of 'correct' formatting (which may be a bug!).

The alternative is to have one long line.